accepter.tf

```
@@ -64,36 +64,31 @@ locals {
 }
 
 # Lookup accepter route tables
-data "aws_route_table" "accepter" {
-  count     = local.accepter_enabled ? local.accepter_subnet_ids_count : 0
-  provider  = aws.accepter
-  subnet_id = element(local.accepter_subnet_ids, count.index)
+data "aws_route_tables" "accepter" {
+  count    = local.count
+  provider = aws.accepter
+  vpc_id   = local.accepter_vpc_id
 }
 
 locals {
-  accepter_aws_route_table_ids           = try(distinct(sort(data.aws_route_table.accepter.*.route_table_id)), [])
+  accepter_aws_route_table_ids           = try(distinct(sort(data.aws_route_tables.accepter[0].ids)), [])
   accepter_aws_route_table_ids_count     = length(local.accepter_aws_route_table_ids)
-  accepter_cidr_block_associations       = try(flatten(data.aws_vpc.accepter.*.cidr_block_associations), [])
+  accepter_cidr_block_associations       = flatten(data.aws_vpc.accepter.*.cidr_block_associations)
   accepter_cidr_block_associations_count = length(local.accepter_cidr_block_associations)
 }
 
 # Create routes from accepter to requester
 resource "aws_route" "accepter" {
-  count                     = local.accepter_enabled ? local.accepter_aws_route_table_ids_count * local.requester_cidr_block_associations_count : 0
+  count                     = module.this.enabled ? local.accepter_aws_route_table_ids_count * local.requester_cidr_block_associations_count : 0
```

```suggestion
  count                     = local.enabled ? local.accepter_aws_route_table_ids_count * local.requester_cidr_block_associations_count : 0
```

why are you changing from `local.accepter_enabled` to the global enabled?